k8s/src/crd.rs

```diff
@@ -0,0 +1,516 @@
+                    if container.image.is_none() {
+                        container.image = Some(String::from(DEFAULT_CPU_IMAGE));
+                    }
+
+                    let env = container
+                        .env
+                        .as_mut()
+                        .expect("no env in a persia podspec template");
+                    env.push(EnvVar {
+                        name: String::from("REPLICA_INDEX"),
+                        value: Some(replica_idx.to_string()),
+                        ..EnvVar::default()
+                    });
+
+                    env.push(EnvVar {
+                        name: String::from("REPLICA_SIZE"),
+                        value: Some(middleware_server.replicas.to_string()),
+                        ..EnvVar::default()
+                    });
+
+                    if let Some(e) = &middleware_server.env {
+                        e.iter().for_each(|env_var| {
+                            env.push(env_var.clone());
+                        });
+                    }
+
+                    pod.metadata.name = Some(get_mid_server_pod_name(job_name, replica_idx));
+                    pod
+                })
+                .collect();
+
+            results.append(&mut middleware_server_spec);
+        }
+
+        if let Some(trainer) = &self.trainer {
+            let mut trainer_spec: Vec<Pod> = (0..trainer.replicas)
+                .into_iter()
+                .map(|replica_idx| {
+                    let mut pod = self.gen_pod_template(job_name, namespace);
+                    let podspec = pod.spec.as_mut().unwrap();
+
+                    let container = podspec.containers.first_mut().unwrap();
+
+                    container.name = "trainer".to_string();
+                    container.args = Some(
+                        vec![
+                            "trainer",
+                            "$(TRAINER_PY_ENTRY_PATH)",
+                            "--nproc-per-node",
+                            "$(NPROC_PER_NODE)",
+                            "--nnodes",
+                            "$(REPLICA_SIZE)",
+                            "--node-rank",
+                            "$(REPLICA_INDEX)",
+                        ]
+                        .into_iter()
+                        .map(|x| x.to_string())
+                        .collect(),
+                    );
+
+                    container.resources = trainer.resources.clone();
+                    container.volume_mounts = trainer.volumeMounts.clone();
+
+                    container.image = trainer.image.clone();
+                    if container.image.is_none() {
+                        container.image = Some(String::from(DEFAULT_CUDA_IMAGE));
+                    }
+
+                    let env = container
+                        .env
+                        .as_mut()
+                        .expect("no env in a persia podspec template");
```

Comment:
for `trainer` or `nn_worker` env spec

Reply:
Fixed in https://github.com/PersiaML/PERSIA/pull/142/commits/4b7070554666fd7c94ef897a0160e2031c1ab7f0